table/server/master/src/main/java/alluxio/master/table/Database.java

```diff
@@ -467,14 +508,55 @@ public boolean hasNext() {
         }
         Table table = mEntry;
         mEntry = null;
-        alluxio.proto.journal.Table.AddTableEntry addTableEntry = table.toJournalProto();
+        alluxio.proto.journal.Table.AddTableEntry addTableEntry = table.getTableJournalProto();
         return Journal.JournalEntry.newBuilder().setAddTable(addTableEntry).build();
       }
 
       @Override
       public void remove() {
         throw new UnsupportedOperationException(
-            "GetTableIteratorr#Iterator#remove is not supported.");
+            "GetTableIterator#Iterator#remove is not supported.");
+      }
+    };
+  }
+
+  private Iterator<Journal.JournalEntry> getTablePartitionsIterator() {
+    final Iterator<Table> it = getTables().iterator();
+    return new Iterator<Journal.JournalEntry>() {
+      private List<alluxio.proto.journal.Table.AddTablePartitionsEntry> mPartitionsEntries
```

Comment:
Instead of 1 table iterator and partition list, should we just have a table iterator, and partition iterator? Everything the partition iterator is empty() we just move on to the next table and get the next partition iterator?

It seems like mixing an iterator and a list is a bit confusing to follow.

Reply:
I changed to add partitionIterator directly in the existing getTableIterator to be cleaner and safer. PTAL